pkg/policies/branch/branch.go

```diff
@@ -568,6 +597,51 @@ func fix(ctx context.Context, rep repositories, c *github.Client,
 				Str("area", polName).
 				Msg("Updated with Fix action.")
 		}
+
+		sp, rsp, err := rep.GetSignaturesProtectedBranch(ctx, owner, repo, b)
+		if err != nil {
+			if rsp != nil && rsp.StatusCode == http.StatusForbidden {
+				log.Warn().
+					Str("org", owner).
+					Str("repo", repo).
+					Str("area", polName).
+					Str("branch", b).
+					Msg("Fix action selected, but cannot access current commit signing settings.")
+				return nil
+			}
+			return err
+		}
+
+		updateRequiredSignatures := mc.RequireSignedCommits != *sp.Enabled
+		if updateRequiredSignatures {
+			var signatureStatus string
+			if mc.RequireSignedCommits && !*sp.Enabled {
+				signatureStatus = "required"
+				_, rsp, err = rep.RequireSignaturesOnProtectedBranch(ctx, owner, repo, b)
+			} else {
+				signatureStatus = "optional"
+				rsp, err = rep.OptionalSignaturesOnProtectedBranch(ctx, owner, repo, b)
```

Comment:
It seemed counter-intuitive to me to write code that would allow **disabling** the requirement of signed commits on protected branches, but if we're going to allow the `requireSignedCommits` setting to be configurable, then it seemed like the code should make sure that branch protection settings are in alignment?